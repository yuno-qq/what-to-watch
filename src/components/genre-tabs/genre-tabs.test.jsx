import React from 'react'
import {describe, test, expect} from "@jest/globals"
import renderer from 'react-test-renderer'

import GenreTabs from "./genre-tabs.jsx"


describe(`<GenreTabs/> snapshots:`, () => {
  test(`should render first tab active`, () => {
    const genres = [
      {
        id: `all`,
        name: `All genres`
      },
      {
        id: `crime`,
        name: `Crime`
      },
      {
        id: `comedy`,
        name: `Comedy`
      },
    ]

    const tree = renderer.create(<GenreTabs
      genres={genres}
      activeGenre={genres[0]}
      onClickTab={() => {}}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test(`should render third tab active`, () => {
    const genres = [
      {
        id: `all`,
        name: `All genres`
      },
      {
        id: `crime`,
        name: `Crime`
      },
      {
        id: `comedy`,
        name: `Comedy`
      },
    ]

    const tree = renderer.create(<GenreTabs
      genres={genres}
      activeGenre={genres[2]}
      onClickTab={() => {}}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
