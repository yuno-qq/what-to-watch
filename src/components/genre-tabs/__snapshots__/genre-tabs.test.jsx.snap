// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreTabs/> snapshots: should render first tab active 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
</ul>
`;

exports[`<GenreTabs/> snapshots: should render third tab active 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
</ul>
`;
